src/schema/v2/users/updateUserSaleProfileMutation.ts

```diff
@@ -0,0 +1,79 @@
+import { GraphQLNonNull, GraphQLString } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import { snakeCase } from "lodash"
+
+interface Input {
+  id: string
+  addressLine1?: string
+  addressLine2?: string
+  city?: string
+  state?: string
+  zip?: string
+  country?: string
+}
+
+interface GravityInput {
+  id: string
+  address_1?: string
+  address_2?: string
+  city?: string
+  state?: string
+  zip?: string
+  country?: string
+}
+
+interface GravityError {
+  statusCode: number
+  body: { error?: string; text?: string; message?: string }
+}
+
+export const updateUserSaleProfileMutation = mutationWithClientMutationId<
+  Input,
+  any | null,
+  ResolverContext
+>({
+  name: "UpdateUserSaleProfileMutation",
+  description: "Update the user sale profile",
+  inputFields: {
+    id: { type: new GraphQLNonNull(GraphQLString) },
+    addressLine1: { type: GraphQLString },
+    addressLine2: { type: GraphQLString },
+    city: { type: GraphQLString },
+    state: { type: GraphQLString },
+    zip: { type: GraphQLString },
+    country: { type: GraphQLString },
+  },
+  outputFields: {},
+  mutateAndGetPayload: async (args, { updateUserSaleProfileLoader }) => {
+    if (!updateUserSaleProfileLoader) {
+      throw new Error(
+        "You need to be signed in as an admin to perform this action"
+      )
+    }
+
+    const initialValue = {} as GravityInput
+    const gravityOptions = Object.keys(args)
+      .filter((key) => key !== "id")
+      .reduce((acc, key) => {
+        if (key === "addressLine1") {
+          return { ...acc, address_1: args[key] }
+        } else if (key === "addressLine2") {
+          return { ...acc, address_2: args[key] }
+        } else {
+          return { ...acc, [snakeCase(key)]: args[key] }
+        }
+      }, initialValue)
+
+    try {
+      return await updateUserSaleProfileLoader?.(args.id, gravityOptions)
+    } catch (err) {
+      if ("body" in (err as any)) {
+        const e = err as GravityError
+        throw new Error(e.body.text ?? e.body.error ?? e.body.message)
+      }
+
+      throw err
+    }
```

Comment:
@jonallured / @dzucconi - since this error handling approach is now being copied all around, can we put it inside a reusable helper (and where would that helper be located)?

Reply:
You might be onto something @damassi 😝 

```
jon@juggernaut:~/code/metaphysics(jonallured/feat-admin-can-manage-sale-profile-data-for-a-user)% ag "interface GravityError"
src/schema/v2/me/unlinkAuthenticationMutation.ts
11:interface GravityError {

src/schema/v2/me/deleteUserInterestMutation.ts
75:interface GravityError {

src/schema/v2/me/linkAuthenticationMutation.ts
18:interface GravityError {

src/schema/v2/me/createUserInterestMutation.ts
87:interface GravityError {

src/schema/v2/me/updateMyPasswordMutation.ts
20:interface GravityError {

src/schema/v2/users/updateUserMutation.ts
22:interface GravityError {

src/schema/v2/users/updateUserSaleProfileMutation.ts
25:interface GravityError {
```

But I'd like to break that off into another PR so I can land this work if that's ok!

Reply:
sounds good!